### ISSUE TITLE
PHPUnit warnings are not handled gracefully

### ISSUE THREAD

**Issue by fzieris** [jakzal/phpunit-globals#17]
With the extension enabled, a test case that would normally lead to a PHPUnit "Warning" results in a PHP "Fatal Error".

See #16 for a test case (with an intentionally flawed data provider) to demonstrate this.

**Expected behavior** (as can be seen without the extension enabled):

```
There was 1 warning:

1) Warning
The data provider specified for Zalas\PHPUnit\Globals\Tests\AnnotationExtensionWithWarningTest::test_it_handles_warnings is invalid.
PHPUnit\Framework\InvalidDataProviderException: The key "test" has already been defined in the data provider "warning_producing_provider".

WARNINGS!
Tests: 1, Assertions: 0, Warnings: 1.
```

**Actual behavior** (with the extension enabled):

```
PHP Fatal error:  Uncaught ReflectionException: Class Warning does not exist in C:\Users\fzieris\Development\phpunit-globals\vendor\phpunit\phpunit\src\Util\Annotation\Registry.php:51
Stack trace:
#0 C:\Users\fzieris\Development\phpunit-globals\vendor\phpunit\phpunit\src\Util\Annotation\Registry.php(51): ReflectionClass->__construct('Warning')
#1 C:\Users\fzieris\Development\phpunit-globals\vendor\phpunit\phpunit\src\Util\Test.php(310): PHPUnit\Util\Annotation\Registry->forClassName('Warning')
#2 C:\Users\fzieris\Development\phpunit-globals\src\AnnotationExtension.php(90): PHPUnit\Util\Test::parseTestMethodAnnotations('Warning')
#3 C:\Users\fzieris\Development\phpunit-globals\src\AnnotationExtension.php(76): Zalas\PHPUnit\Globals\AnnotationExtension->parseTestMethodAnnotations('Warning')
#4 C:\Users\fzieris\Development\phpunit-globals\src\AnnotationExtension.php(71): Zalas\PHPUnit\Globals\AnnotationExtension->findGlobalVarAnnotations('Warning')
#5 C:\Users\fzieris\Development\phpunit-globals\src\AnnotationExtension.php(49): Zalas\PHPUnit\Gl 
in C:\Users\fzieris\Development\phpunit-globals\vendor\phpunit\phpunit\src\Util\Annotation\Registry.php on line 54
````

(The stacktrace appears to be truncated for some unclear reason.)

**Comment by jakzal** [comment]
Interesting. Thanks for reporting! 

Seems that the extension is called with a `Warning` as an argument. Very odd.

**Comment by jakzal** [comment]
Fixed in #19
